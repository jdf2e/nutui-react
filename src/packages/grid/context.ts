import React from 'react'
import { GridItemProps } from '../griditem/types'

const gridContext = {
  onClick: (item: GridItemProps, index: number) => {},
}

export default React.createContext(gridContext)
