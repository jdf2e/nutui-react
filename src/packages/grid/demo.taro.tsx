import React from 'react'
import Taro from '@tarojs/taro'
import { Dongdong, My } from '@nutui/icons-react-taro'
import { useTranslate } from '@/sites/assets/locale/taro'
import { Avatar, Grid } from '@/packages/nutui.react.taro'
import Header from '@/sites/components/header'

interface T {
  basic: string
  text: string
  columnNum: string
  square: string
  gutter: string
  reverse: string
  horizontal: string
  iconStyle: string
  customContent: string
  event: string
}
const GridDemo = () => {
  const [translated] = useTranslate<T>({
    'zh-CN': {
      basic: '基本用法',
      text: '文字',
      columnNum: '自定义列数',
      square: '正方形格子',
      gutter: '格子间距',
      reverse: '内容翻转',
      horizontal: '内容横向',
      iconStyle: '图标颜色/大小',
      customContent: '自定义内容',
      event: '点击子项事件',
    },
    'zh-TW': {
      basic: '基本用法',
      text: '文字',
      columnNum: '自定義列數',
      square: '正方形格子',
      gutter: '格子間距',
      reverse: '內容翻轉',
      horizontal: '內容橫向',
      iconStyle: '圖標顏色/大小',
      customContent: '自定義內容',
      event: '點擊子項事件',
    },
    'en-US': {
      basic: 'Basic Usage',
      text: 'text',
      columnNum: 'Column Num',
      square: 'Square',
      gutter: 'Gutter',
      reverse: 'Reverse',
      horizontal: 'Horizontal',
      iconStyle: 'Icon Style',
      customContent: 'Custom Content',
      event: 'Grid Item Click',
    },
  })

  const handleClick = () => {
    // Toast.text('点击了第几个')
    Taro.showToast({ title: '点击了第几个' })
  }

  const onClick = (item: any, index: number) => {
    Taro.showToast({ title: `点击了${item.text}，第${index}个` })
  }

  return (
    <>
      <Header />
      <div className={`demo ${Taro.getEnv() === 'WEB' ? 'web' : ''}`}>
        <h2>{translated.basic}</h2>
        <Grid>
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
        </Grid>

        <h2>{translated.columnNum}</h2>
        <Grid columnNum={3}>
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
        </Grid>

        <h2>{translated.square}</h2>
        <Grid columnNum={3} square>
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
        </Grid>

        <h2>{translated.gutter}</h2>
        <Grid gutter={3}>
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
        </Grid>

        <h2>{translated.reverse}</h2>
        <Grid reverse>
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
        </Grid>

        <h2>{translated.horizontal}</h2>
        <Grid direction="horizontal">
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
        </Grid>

        <h2>{translated.iconStyle}</h2>
        <Grid columnNum="3">
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item
            icon={<Dongdong size={40} color="#478EF2" />}
            text={translated.text}
            color="#478EF2"
          />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
        </Grid>

        <h2>{translated.customContent}</h2>
        <Grid border={false}>
          <Grid.Item icon={<Dongdong />} text={<span>More</span>} />
          <Grid.Item
            text={
              <Avatar
                className="demo-avatar"
                icon={<My color="#fff" />}
                bgColor="#FA2C19"
              />
            }
            onClick={handleClick}
          />
          <Grid.Item
            icon={
              <Avatar
                className="demo-avatar"
                icon={<My color="#fff" />}
                bgColor="#FA2C19"
              />
            }
          />
          <Grid.Item>
            <Avatar
              size="large"
              icon={
                <img
                  width="100%"
                  height="100%"
                  alt="头像"
                  src="https://img12.360buyimg.com/imagetools/jfs/t1/143702/31/16654/116794/5fc6f541Edebf8a57/4138097748889987.png"
                />
              }
            />
          </Grid.Item>
        </Grid>

        <h2>{translated.event}</h2>
        <Grid direction="horizontal" onClick={onClick}>
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
          <Grid.Item icon={<Dongdong />} text={translated.text} />
        </Grid>
      </div>
    </>
  )
}

export default GridDemo
