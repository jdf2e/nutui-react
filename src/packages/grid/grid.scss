@import '../griditem/griditem.scss';

.nut-grid {
  display: flex;
  flex-wrap: wrap;
  border: 0 solid $grid-border-color;

  &__border {
    border-top-width: 1px;
    border-left-width: 1px;
  }
}
