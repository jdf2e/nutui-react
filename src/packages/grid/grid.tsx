import React, { CSSProperties, FunctionComponent } from 'react'
import classNames from 'classnames'
import { GridItem, GridItemProps } from '../griditem/griditem'
import GridContext from './context'
import { pxCheck } from '@/utils/px-check'

export type GridDirection = 'horizontal' | 'vertical'

export interface GridProps {
  columns: string | number
  gap: string | number
  center: boolean
  square: boolean
  reverse: boolean
  direction: GridDirection
  className?: string
  style?: CSSProperties
  onClick: (item: GridItemProps, index: number) => void
}

const defaultProps = {
  columns: 4,
  gap: 0,
  center: true,
  square: false,
  reverse: false,
  direction: 'vertical',
} as GridProps

export const Grid: FunctionComponent<
  Partial<GridProps> & Omit<React.HTMLAttributes<HTMLDivElement>, 'onClick'>
> & {
  Item: typeof GridItem
} = (props) => {
  const {
    children,
    columns,
    gap,
    center,
    square,
    reverse,
    direction,
    style,
    className,
    onClick,
    ...rest
  } = { ...defaultProps, ...props }
  const childrenDom = React.Children.toArray(children)

  const classPrefix = 'nut-grid'

  const rootClass = () => {
    return classNames(
      classPrefix,
      {
        [`${classPrefix}-border`]: !gap,
      },
      className
    )
  }

  const rootStyle = () => {
    let styleSelf: CSSProperties = {}
    if (style) {
      styleSelf = style
    }
    if (gap) {
      styleSelf.paddingLeft = pxCheck(gap)
    }

    return styleSelf
  }

  return (
    <div className={rootClass()} style={rootStyle()} {...rest}>
      <GridContext.Provider value={{ onClick }}>
        {childrenDom.map((item: any, idex: number) => {
          return React.cloneElement(item, {
            index: idex,
            columns,
            center,
            gap,
            square,
            reverse,
            direction,
          })
        })}
      </GridContext.Provider>
    </div>
  )
}

Grid.displayName = 'NutGrid'
Grid.Item = GridItem
