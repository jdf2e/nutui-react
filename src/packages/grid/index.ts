import { Grid } from './grid'

export type { GridDirection } from './grid'
export default Grid
