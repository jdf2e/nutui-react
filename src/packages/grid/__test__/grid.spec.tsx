import * as React from 'react'

import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import { Grid } from '../grid'
import { GridItem } from '../../griditem/griditem'
import Avatar from '@/packages/avatar'

test('should match snapshot', () => {
  const { container } = render(
    <>
      <Grid>
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
      </Grid>
    </>
  )
  expect(container).toMatchSnapshot()
})

test('render with column num', () => {
  const { container } = render(
    <>
      <Grid columnNum={3}>
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
      </Grid>
    </>
  )

  const regionItem = container.querySelector('.nut-grid-item')
  expect(regionItem).toHaveStyle('flex-basis:33.333333333333336%')
})

test('render with props gutter', () => {
  const { container } = render(
    <>
      <Grid gutter={3}>
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
      </Grid>
    </>
  )

  const regionItem = container.querySelector('.nut-grid-item')
  expect(regionItem).toHaveStyle('padding-right:3px')
})

test('render with props reverse', () => {
  const { container } = render(
    <>
      <Grid reverse>
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
      </Grid>
    </>
  )

  const regionItem = container.querySelector('.nut-grid-item')
  const item = regionItem?.querySelector('.nut-grid-item__content')
  expect(item).toHaveClass('nut-grid-item__content--reverse')
})

test('render with props direction horizontal', () => {
  const { container } = render(
    <>
      <Grid direction="horizontal">
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
      </Grid>
    </>
  )

  const regionItem = container.querySelector('.nut-grid-item')
  const item = regionItem?.querySelector('.nut-grid-item__content')
  expect(item).toHaveClass('nut-grid-item__content--horizontal')
})

test('render with custom', () => {
  const { container } = render(
    <>
      <Grid columnNum="3" iconColor="#fa2c19">
        <GridItem icon="dongdong" text="文本" />
        <GridItem
          icon="dongdong"
          text="文本"
          iconColor="#478EF2"
          iconSize="40"
        />
        <GridItem icon="dongdong" text="文本" />
      </Grid>
    </>
  )

  const gridItem = container.querySelectorAll('.nut-grid-item')
  expect(gridItem.length).toBe(3)
  const one = gridItem[0].querySelector('.nutui-iconfont')
  const two = gridItem[1].querySelector('.nutui-iconfont')
  expect(one).toHaveStyle('color: rgb(250, 44, 25)')
  expect(two).toHaveStyle(
    'color: rgb(71, 142, 242); font-size: 40px; width: 40px; height: 40px;'
  )
})

test('render with custom content', () => {
  const { container } = render(
    <>
      <Grid direction="horizontal">
        <GridItem icon="dongdong" text={<span>More</span>} />
        <GridItem
          text={
            <Avatar
              className="demo-avatar"
              icon="my"
              color="#fff"
              bgColor="#FA2C19"
            />
          }
        />
        <GridItem
          icon={
            <Avatar
              className="demo-avatar"
              icon="my"
              color="#fff"
              bgColor="#FA2C19"
            />
          }
        />
        <GridItem>
          <Avatar
            size="large"
            icon="https://img12.360buyimg.com/imagetools/jfs/t1/143702/31/16654/116794/5fc6f541Edebf8a57/4138097748889987.png"
          />
        </GridItem>
      </Grid>
    </>
  )

  expect(container).toMatchSnapshot()
})

test('render with props center', () => {
  const { container } = render(
    <>
      <Grid center>
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
        <GridItem icon="dongdong" text="文本" />
      </Grid>
    </>
  )

  const regionItem = container.querySelector('.nut-grid-item')
  const item = regionItem?.querySelector('.nut-grid-item__content')
  expect(item).toHaveClass('nut-grid-item__content--center')
})
