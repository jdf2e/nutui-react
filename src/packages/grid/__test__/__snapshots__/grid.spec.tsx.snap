// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render with custom content 1`] = `
<div>
  <div
    class="nut-grid nut-grid__border"
  >
    <div
      class="nut-grid-item"
      iconclassprefix="nut-icon"
      iconfontclassname="nutui-iconfont"
      style="flex-basis: 25%;"
    >
      <div
        class="nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center nut-grid-item__content--horizontal"
      >
        <svg
          aria-labelledby=""
          class="nut-icon"
          role="presentation"
          viewBox="0 0 1079 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M802.595 1018.465h-11.07c-58.12-8.303-119.006-22.14-182.66-47.049l-11.07-5.535h-11.07c-16.606 0-30.444 2.768-47.05 2.768C240.779 968.649 0 750.01 0 484.324S240.778 0 539.676 0s539.675 218.638 539.675 484.324c0 152.217-80.26 298.898-218.637 390.227l-2.768 2.768v88.562c0 27.676-24.908 52.584-55.351 52.584zm-202.033-135.61c5.535 0 13.838 0 19.373 2.767l19.373 8.302c47.049 16.606 94.097 30.444 135.61 38.746v-71.956c0-19.373 11.071-35.979 27.677-47.05 5.535-2.767 11.07-5.534 13.837-8.302 116.238-77.492 179.892-193.73 179.892-321.038 0-221.405-204.8-401.297-456.648-401.297S83.027 262.919 83.027 484.324s204.8 401.298 456.649 401.298c13.838 0 27.675 0 41.513-2.768h19.373zm243.546 0zm-304.432-398.53c77.492 0 138.378-24.909 138.378-55.352s-60.886-41.514-138.378-41.514-138.379 11.07-138.379 41.514 60.887 55.351 138.379 55.351zm-226.94-185.428c19.372 11.07 27.675 35.979 13.837 55.352l-27.676 47.048c-11.07 19.373-35.978 27.676-55.351 13.838-19.373-11.07-27.676-35.978-13.838-55.351l27.676-47.049c11.07-19.373 35.978-27.676 55.351-13.838zm456.648 0c-19.373 11.07-27.676 35.979-13.838 55.352l27.676 47.048c11.07 19.373 35.978 27.676 55.351 13.838 19.373-11.07 27.676-35.978 13.838-55.351l-27.676-47.049c-11.07-19.373-35.978-27.676-55.351-13.838zM539.676 705.73c-138.379 0-268.454-58.12-351.481-157.752-11.07-11.07-8.303-30.443 2.767-38.746 11.07-11.07 30.443-8.302 38.746 2.768 74.724 88.562 188.195 138.378 309.968 138.378 121.773 0 240.778-52.583 312.735-141.146 11.07-11.07 27.675-13.837 38.746-2.767 11.07 11.07 13.838 27.676 2.767 38.746-83.027 99.632-215.87 160.519-354.248 160.519z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
        <div
          class="nut-grid-item__text"
        >
          <span>
            More
          </span>
        </div>
      </div>
    </div>
    <div
      class="nut-grid-item"
      iconclassprefix="nut-icon"
      iconfontclassname="nutui-iconfont"
      style="flex-basis: 25%;"
    >
      <div
        class="nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center nut-grid-item__content--horizontal"
      >
        <div
          class="nut-grid-item__text"
        >
          <div
            class="nut-avatar nut-avatar-normal nut-avatar-round demo-avatar"
            iconclassprefix="nut-icon"
            iconfontclassname="nutui-iconfont"
            style="background-color: rgb(250, 44, 25); color: rgb(102, 102, 102);"
          >
            <svg
              aria-labelledby=""
              class=" icon nut-icon"
              color="#fff"
              role="presentation"
              viewBox="0 0 1024 1024"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M512 813.949c-225.805 0-406.974-183.795-406.974-406.975S286.195 0 512 0s406.974 181.17 406.974 406.974S735.18 813.95 512 813.95zm0-735.18c-181.17 0-328.205 147.036-328.205 328.205S330.83 735.18 512 735.18s328.205-147.035 328.205-328.205S693.17 78.77 512 78.77zm202.174 939.98c-21.005 0-39.384-18.38-39.384-39.385 0-89.272-73.518-162.79-162.79-162.79-89.272 0-162.79 73.518-162.79 162.79 0 21.005-18.38 39.385-39.384 39.385s-39.385-18.38-39.385-39.385c0-133.908 107.651-241.559 241.559-241.559s241.559 107.651 241.559 241.56c0 21.004-15.754 39.384-39.385 39.384zM512 525.129c-39.385 0-78.77-13.129-105.026-36.76-15.753-13.128-18.38-39.384-2.625-55.138 13.128-15.754 39.384-18.38 55.138-2.626 23.63 21.005 81.395 21.005 105.026 0 15.754-13.128 42.01-13.128 55.138 5.251 13.128 15.754 13.128 42.01-5.251 55.139C590.77 512 551.385 525.128 512 525.128z"
                fill="currentColor"
                fill-opacity="0.9"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="nut-grid-item"
      iconclassprefix="nut-icon"
      iconfontclassname="nutui-iconfont"
      style="flex-basis: 25%;"
    >
      <div
        class="nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center nut-grid-item__content--horizontal"
      >
        <div
          class="nut-avatar nut-avatar-normal nut-avatar-round demo-avatar"
          iconclassprefix="nut-icon"
          iconfontclassname="nutui-iconfont"
          style="background-color: rgb(250, 44, 25); color: rgb(102, 102, 102);"
        >
          <svg
            aria-labelledby=""
            class=" icon nut-icon"
            color="#fff"
            role="presentation"
            viewBox="0 0 1024 1024"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 813.949c-225.805 0-406.974-183.795-406.974-406.975S286.195 0 512 0s406.974 181.17 406.974 406.974S735.18 813.95 512 813.95zm0-735.18c-181.17 0-328.205 147.036-328.205 328.205S330.83 735.18 512 735.18s328.205-147.035 328.205-328.205S693.17 78.77 512 78.77zm202.174 939.98c-21.005 0-39.384-18.38-39.384-39.385 0-89.272-73.518-162.79-162.79-162.79-89.272 0-162.79 73.518-162.79 162.79 0 21.005-18.38 39.385-39.384 39.385s-39.385-18.38-39.385-39.385c0-133.908 107.651-241.559 241.559-241.559s241.559 107.651 241.559 241.56c0 21.004-15.754 39.384-39.385 39.384zM512 525.129c-39.385 0-78.77-13.129-105.026-36.76-15.753-13.128-18.38-39.384-2.625-55.138 13.128-15.754 39.384-18.38 55.138-2.626 23.63 21.005 81.395 21.005 105.026 0 15.754-13.128 42.01-13.128 55.138 5.251 13.128 15.754 13.128 42.01-5.251 55.139C590.77 512 551.385 525.128 512 525.128z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      class="nut-grid-item"
      iconclassprefix="nut-icon"
      iconfontclassname="nutui-iconfont"
      style="flex-basis: 25%;"
    >
      <div
        class="nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center nut-grid-item__content--horizontal"
      >
        <div
          class="nut-avatar nut-avatar-large nut-avatar-round"
          iconclassprefix="nut-icon"
          iconfontclassname="nutui-iconfont"
          style="background-color: rgb(238, 238, 238); color: rgb(102, 102, 102);"
        >
          <div
            class="nut-image   icon"
          >
            <img
              alt=""
              class="nut-img"
              src="https://img12.360buyimg.com/imagetools/jfs/t1/143702/31/16654/116794/5fc6f541Edebf8a57/4138097748889987.png"
              style="object-fit: fill; object-position: center;"
            />
            <div
              class="nut-img-loading"
            >
              <svg
                aria-labelledby=""
                class="nut-icon"
                role="presentation"
                viewBox="0 0 1024 1024"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M853.161 892.55H362.595l-209.433-.414a18.5 18.5 0 0 1-1.813-.092c-24.849-2.485-47.825-14.06-64.696-32.595-16.991-18.665-26.35-42.855-26.35-68.112V264.626c0-55.808 45.404-101.211 101.212-101.211H853.16c55.808 0 101.21 45.403 101.21 101.211v225.513c0 .275-.006.552-.018.826-.022.494-1.972 51.723 15.481 85.462 4.717 9.119 1.148 20.335-7.97 25.053-9.117 4.714-20.335 1.15-25.053-7.97-21.3-41.177-19.845-97.643-19.619-103.74V264.627c0-35.307-28.724-64.031-64.03-64.031H161.515c-35.308 0-64.032 28.724-64.032 64.031v526.711c0 32.755 24.321 59.958 56.718 63.62l208.431.412h490.53c35.306 0 64.03-28.725 64.03-64.032l-.382-93.677c-.105-1.75-1.587-19.548-19.55-42.5a9.548 9.548 0 0 1-.147-.19l-21.712-28.489-117.227-155.119c-.416-.55-38.995-50.852-86.272-45.534-38.335 4.315-75.955 45.164-108.79 118.132-17.615 39.141-34.65 68.269-52.082 89.046-17.607 20.986-35.68 33.52-55.251 38.317-43.423 10.638-81.05-18.995-120.887-50.373l-5.057-3.98c-46.555-36.578-68.75-28.224-158.33 59.602-7.33 7.188-19.1 7.071-26.289-.26-7.187-7.33-7.07-19.1.26-26.287 46.659-45.746 77.544-72.727 107.086-84.283 33.358-13.048 64.275-6.267 100.242 21.994l5.093 4.008c33.923 26.72 63.22 49.795 89.029 43.467 25.618-6.279 53.3-42.114 82.28-106.509 39.14-86.976 85.838-134.027 138.797-139.849 68.454-7.515 117.824 57.405 119.892 60.172l117.122 154.98 21.6 28.344c26.277 33.631 27.334 61.639 27.367 64.725l.001.128.384 94.06c.001 55.884-45.4 101.287-101.209 101.287z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
                ,
                <path
                  d="M312.328 446.968c-42.325 0-76.759-34.434-76.759-76.76s34.434-76.759 76.76-76.759 76.759 34.435 76.759 76.76-34.434 76.759-76.76 76.759zm0-116.34c-21.824 0-39.579 17.756-39.579 39.58s17.755 39.58 39.58 39.58 39.579-17.756 39.579-39.58-17.755-39.58-39.58-39.58z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`should match snapshot 1`] = `
<div>
  <div
    class="nut-grid nut-grid__border"
  >
    <div
      class="nut-grid-item"
      iconclassprefix="nut-icon"
      iconfontclassname="nutui-iconfont"
      style="flex-basis: 25%;"
    >
      <div
        class="nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center nut-grid-item__content--vertical"
      >
        <svg
          aria-labelledby=""
          class="nut-icon"
          role="presentation"
          viewBox="0 0 1079 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M802.595 1018.465h-11.07c-58.12-8.303-119.006-22.14-182.66-47.049l-11.07-5.535h-11.07c-16.606 0-30.444 2.768-47.05 2.768C240.779 968.649 0 750.01 0 484.324S240.778 0 539.676 0s539.675 218.638 539.675 484.324c0 152.217-80.26 298.898-218.637 390.227l-2.768 2.768v88.562c0 27.676-24.908 52.584-55.351 52.584zm-202.033-135.61c5.535 0 13.838 0 19.373 2.767l19.373 8.302c47.049 16.606 94.097 30.444 135.61 38.746v-71.956c0-19.373 11.071-35.979 27.677-47.05 5.535-2.767 11.07-5.534 13.837-8.302 116.238-77.492 179.892-193.73 179.892-321.038 0-221.405-204.8-401.297-456.648-401.297S83.027 262.919 83.027 484.324s204.8 401.298 456.649 401.298c13.838 0 27.675 0 41.513-2.768h19.373zm243.546 0zm-304.432-398.53c77.492 0 138.378-24.909 138.378-55.352s-60.886-41.514-138.378-41.514-138.379 11.07-138.379 41.514 60.887 55.351 138.379 55.351zm-226.94-185.428c19.372 11.07 27.675 35.979 13.837 55.352l-27.676 47.048c-11.07 19.373-35.978 27.676-55.351 13.838-19.373-11.07-27.676-35.978-13.838-55.351l27.676-47.049c11.07-19.373 35.978-27.676 55.351-13.838zm456.648 0c-19.373 11.07-27.676 35.979-13.838 55.352l27.676 47.048c11.07 19.373 35.978 27.676 55.351 13.838 19.373-11.07 27.676-35.978 13.838-55.351l-27.676-47.049c-11.07-19.373-35.978-27.676-55.351-13.838zM539.676 705.73c-138.379 0-268.454-58.12-351.481-157.752-11.07-11.07-8.303-30.443 2.767-38.746 11.07-11.07 30.443-8.302 38.746 2.768 74.724 88.562 188.195 138.378 309.968 138.378 121.773 0 240.778-52.583 312.735-141.146 11.07-11.07 27.675-13.837 38.746-2.767 11.07 11.07 13.838 27.676 2.767 38.746-83.027 99.632-215.87 160.519-354.248 160.519z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
        <div
          class="nut-grid-item__text"
        >
          文本
        </div>
      </div>
    </div>
    <div
      class="nut-grid-item"
      iconclassprefix="nut-icon"
      iconfontclassname="nutui-iconfont"
      style="flex-basis: 25%;"
    >
      <div
        class="nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center nut-grid-item__content--vertical"
      >
        <svg
          aria-labelledby=""
          class="nut-icon"
          role="presentation"
          viewBox="0 0 1079 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M802.595 1018.465h-11.07c-58.12-8.303-119.006-22.14-182.66-47.049l-11.07-5.535h-11.07c-16.606 0-30.444 2.768-47.05 2.768C240.779 968.649 0 750.01 0 484.324S240.778 0 539.676 0s539.675 218.638 539.675 484.324c0 152.217-80.26 298.898-218.637 390.227l-2.768 2.768v88.562c0 27.676-24.908 52.584-55.351 52.584zm-202.033-135.61c5.535 0 13.838 0 19.373 2.767l19.373 8.302c47.049 16.606 94.097 30.444 135.61 38.746v-71.956c0-19.373 11.071-35.979 27.677-47.05 5.535-2.767 11.07-5.534 13.837-8.302 116.238-77.492 179.892-193.73 179.892-321.038 0-221.405-204.8-401.297-456.648-401.297S83.027 262.919 83.027 484.324s204.8 401.298 456.649 401.298c13.838 0 27.675 0 41.513-2.768h19.373zm243.546 0zm-304.432-398.53c77.492 0 138.378-24.909 138.378-55.352s-60.886-41.514-138.378-41.514-138.379 11.07-138.379 41.514 60.887 55.351 138.379 55.351zm-226.94-185.428c19.372 11.07 27.675 35.979 13.837 55.352l-27.676 47.048c-11.07 19.373-35.978 27.676-55.351 13.838-19.373-11.07-27.676-35.978-13.838-55.351l27.676-47.049c11.07-19.373 35.978-27.676 55.351-13.838zm456.648 0c-19.373 11.07-27.676 35.979-13.838 55.352l27.676 47.048c11.07 19.373 35.978 27.676 55.351 13.838 19.373-11.07 27.676-35.978 13.838-55.351l-27.676-47.049c-11.07-19.373-35.978-27.676-55.351-13.838zM539.676 705.73c-138.379 0-268.454-58.12-351.481-157.752-11.07-11.07-8.303-30.443 2.767-38.746 11.07-11.07 30.443-8.302 38.746 2.768 74.724 88.562 188.195 138.378 309.968 138.378 121.773 0 240.778-52.583 312.735-141.146 11.07-11.07 27.675-13.837 38.746-2.767 11.07 11.07 13.838 27.676 2.767 38.746-83.027 99.632-215.87 160.519-354.248 160.519z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
        <div
          class="nut-grid-item__text"
        >
          文本
        </div>
      </div>
    </div>
    <div
      class="nut-grid-item"
      iconclassprefix="nut-icon"
      iconfontclassname="nutui-iconfont"
      style="flex-basis: 25%;"
    >
      <div
        class="nut-grid-item__content nut-grid-item__content--border nut-grid-item__content--center nut-grid-item__content--vertical"
      >
        <svg
          aria-labelledby=""
          class="nut-icon"
          role="presentation"
          viewBox="0 0 1079 1024"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M802.595 1018.465h-11.07c-58.12-8.303-119.006-22.14-182.66-47.049l-11.07-5.535h-11.07c-16.606 0-30.444 2.768-47.05 2.768C240.779 968.649 0 750.01 0 484.324S240.778 0 539.676 0s539.675 218.638 539.675 484.324c0 152.217-80.26 298.898-218.637 390.227l-2.768 2.768v88.562c0 27.676-24.908 52.584-55.351 52.584zm-202.033-135.61c5.535 0 13.838 0 19.373 2.767l19.373 8.302c47.049 16.606 94.097 30.444 135.61 38.746v-71.956c0-19.373 11.071-35.979 27.677-47.05 5.535-2.767 11.07-5.534 13.837-8.302 116.238-77.492 179.892-193.73 179.892-321.038 0-221.405-204.8-401.297-456.648-401.297S83.027 262.919 83.027 484.324s204.8 401.298 456.649 401.298c13.838 0 27.675 0 41.513-2.768h19.373zm243.546 0zm-304.432-398.53c77.492 0 138.378-24.909 138.378-55.352s-60.886-41.514-138.378-41.514-138.379 11.07-138.379 41.514 60.887 55.351 138.379 55.351zm-226.94-185.428c19.372 11.07 27.675 35.979 13.837 55.352l-27.676 47.048c-11.07 19.373-35.978 27.676-55.351 13.838-19.373-11.07-27.676-35.978-13.838-55.351l27.676-47.049c11.07-19.373 35.978-27.676 55.351-13.838zm456.648 0c-19.373 11.07-27.676 35.979-13.838 55.352l27.676 47.048c11.07 19.373 35.978 27.676 55.351 13.838 19.373-11.07 27.676-35.978 13.838-55.351l-27.676-47.049c-11.07-19.373-35.978-27.676-55.351-13.838zM539.676 705.73c-138.379 0-268.454-58.12-351.481-157.752-11.07-11.07-8.303-30.443 2.767-38.746 11.07-11.07 30.443-8.302 38.746 2.768 74.724 88.562 188.195 138.378 309.968 138.378 121.773 0 240.778-52.583 312.735-141.146 11.07-11.07 27.675-13.837 38.746-2.767 11.07 11.07 13.838 27.676 2.767 38.746-83.027 99.632-215.87 160.519-354.248 160.519z"
            fill="currentColor"
            fill-opacity="0.9"
          />
        </svg>
        <div
          class="nut-grid-item__text"
        >
          文本
        </div>
      </div>
    </div>
  </div>
</div>
`;
